substrate/frame/revive/src/storage/meter.rs

```diff
@@ -372,6 +380,27 @@ where
 			Origin::Root => return Ok(Deposit::Charge(Zero::zero())),
 			Origin::Signed(o) => o,
 		};
+		self.charges.sort_by(|a, b| a.contract.cmp(&b.contract));
```

Comment:
Are we sure it's safe to sort? if so can we add a comment that justify it

Reply:
ah got it and we always process the refunds first as well below.

Reply:
if we are doing these changes, I guess we could just use  a btreemap
or even change the Vec<Charge<T>> to  a BtreeMap<T::AccountId, Charge<T>> directly and do it when we absorb charges directly.

@athei  thoughts?